setup.py

```diff
@@ -1,7 +1,71 @@
 import os.path
 import sys
+import warnings
+import platform
 
 import setuptools
+from distutils.core import Extension
+
+# The following code is copied from
+# https://github.com/mongodb/mongo-python-driver/blob/master/setup.py
+# to support installing without the extension on platforms where
+# no compiler is available.
+from distutils.command.build_ext import build_ext
+
+
+class custom_build_ext(build_ext):
+    """Allow C extension building to fail.
+    The C extension speeds up websocket masking, but is not essential.
+    """
+
+    warning_message = """
+********************************************************************
+WARNING: %s could not
+be compiled. No C extensions are essential for websockets to run,
+although they do result in significant speed improvements for
+websockets.
+%s
+Here are some hints for popular operating systems:
+If you are seeing this message on Linux you probably need to
+install GCC and/or the Python development package for your
+version of Python.
+Debian and Ubuntu users should issue the following command:
+    $ sudo apt-get install build-essential python-dev
+RedHat and CentOS users should issue the following command:
+    $ sudo yum install gcc python-devel
+Fedora users should issue the following command:
+    $ sudo dnf install gcc python-devel
+If you are seeing this message on OSX please read the documentation
+here:
+http://api.mongodb.org/python/current/installation.html#osx
```

Comment:
I really, really don't want to handle support questions about MongoDB's documentation.